docs/configuration/config-file-reference.md

```diff
@@ -4245,6 +4254,21 @@ ring:
 # CLI flag: -experimental.ruler.enable-api
 [enable_api: <boolean> | default = false]
 
+# Enable rulers to store a copy of rules owned by other rulers with default
+# state (state before any evaluation) and send this copy in list API requests as
+# backup in case the ruler who owns the rule fails to send its rules. This
+# allows the rules API to handle ruler outage by returning rules with default
+# state. Ring replication-factor needs to be set to 3 or more for this to be
+# useful.
+# CLI flag: -ruler.api-enable-rules-backup
+[api_enable_rules_backup: <boolean> | default = false]
+
+# Remove duplicate rules in the prometheus rules and alerts API response. If
+# there are duplicate rules the rule with the latest evaluation timestamp will
+# be kept.
+# CLI flag: -ruler.api-deduplicate-rules
+[api_deduplicate_rules: <boolean> | default = false]
```

Comment:
I gathered that this config is for transition purposes when an user upgrade to a version of Cortex that supports replication. 

In this cause I think the we should make the purpose more explicit, and call out this flag is subject to removal. Maybe we should event mark this field as "experimental" to tag along the "experimental" SLA of Cortex.

Reply:
It is but also not 100%. During ring change there is a chance that a rule group is loaded by multiple rulers just because of the timing of the sync rules and the ring propagation delay. So this flag can be used to avoid duplicate rule groups in the list during these events.